persistence/javadsl/src/main/scala/com/lightbend/lagom/javadsl/persistence/PersistentEntity.scala

```diff
@@ -191,19 +191,17 @@ abstract class PersistentEntity[Command, Event, State] {
      * Create a `BehaviorBuilder` that corresponds to this `Behavior`, i.e. the builder
      * is populated with same state, same event and command handler functions.
      */
-    def builder(): BehaviorBuilder = new BehaviorBuilder(state, eventHandlers, commandHandlers)
+    def builder(): BehaviorBuilder = new BehaviorBuilderImpl(state, eventHandlers, commandHandlers)
 
   }
 
-  /**
-   * Mutable builder that is used for defining the event and command handlers.
-   * Use [#build] to create the immutable [[Behavior]].
-   */
-  protected final class BehaviorBuilder(
+  // in order to keep all the constructors protected but still generate javadocs this class
+  // extends a public sealed trait where documentation is provided.
```

Comment:
It's a shame that all of this is necessary... in Java it would be simple to make the class public and the constructors private ☹️

Reply:
Class public and constructors private is:

```scala
final class BehaviorBuilder private (...) {
    private this(...) = ???
}
```